packaging/os/yum.py

```diff
@@ -544,7 +546,7 @@ def install(module, items, repoq, yum_basecmd, conf_file, en_repos, dis_repos):
                     f.close()
                     pkg = package
                 except Exception, e:
-                    shutil.rmtree(tempdir)
+                    delete_temporary_directory(tempdir)
```

Comment:
This is a change of behaviour in the case that the download failed AND the directory couldn't be removed the download error would be "hidden" behind the directory removal error. I'm 20% thinking this is ok, 20% thinking this is wrong but 60% thinking that it's such a small probability that it actually happens it's not been worth writing this out.